tests/bluetooth/host/keys/bt_keys_get_addr/src/main.c

```diff
@@ -0,0 +1,173 @@
+	zassert_true(returned_key->id == id,
+		     "bt_keys_get_addr() returned a reference with an incorrect ID");
+	zassert_true(!bt_addr_le_cmp(&returned_key->addr, addr),
+		     "bt_keys_get_addr() set incorrect address %s value, expected %s",
+		     bt_addr_le_str(&returned_key->addr), bt_addr_le_str(addr));
+	zassert_false(check_key_pool_is_empty(),
+		      "List is empty, (ID, Address) pair insertion failed");
+}
+
+/*
+ *  Test no equal references returned by bt_keys_get_addr()
+ *
+ *  Constraints:
+ *   - Keys pool has been filled
+ *
+ *  Expected behaviour:
+ *   - All returned references are different from each other
+ */
+void test_no_equal_references(void)
+{
+	struct bt_keys *keys_pool = bt_keys_get_key_pool();
+
+	for (uint32_t i = 0; i < ARRAY_SIZE(returned_keys_refs); i++) {
+		struct bt_keys *returned_ref = returned_keys_refs[i];
+
+		zassert_equal_ptr(keys_pool + i, returned_ref,
+				  "bt_keys_get_addr() returned unexpected reference at slot %u", i);
+	}
+}
+
+/*
+ *  Test getting a valid key reference by a matching ID and address
+ *
+ *  Constraints:
+ *   - ID and address pair has been inserted in the list
+ *
+ *  Expected behaviour:
+ *   - A valid reference is returned by bt_keys_get_addr() that
+ *     matches the one returned after adding the ID and address pair
+ *   - ID value matches the one passed to bt_keys_get_addr()
+ *   - Address value matches the one passed to bt_keys_get_addr()
+ */
+void test_get_key_by_matched_id_and_address(void)
+{
+	struct bt_keys *returned_key, *expected_key_ref;
+	uint8_t id = current_params_vector->id;
+	bt_addr_le_t *addr = current_params_vector->addr;
+
+	returned_key = bt_keys_get_addr(id, addr);
+	/* 'params_it' is incremented before running the test case */
+	expected_key_ref = returned_keys_refs[params_it - 1];
+
+	zassert_true(returned_key != NULL, "bt_keys_get_addr() returned a non-valid reference");
+	zassert_equal_ptr(returned_key, expected_key_ref,
+			  "bt_keys_get_addr() returned unexpected reference");
+}
+
+/* Initialize test variables */
+static void reset_global_test_round_params(void)
+{
+	params_it = 0;
+	current_params_vector = NULL;
+}
+
+/* Setup test variables */
+static void unit_test_setup(void)
+{
+	zassert_true((params_it < (ARRAY_SIZE(testing_id_addr_pair_lut))),
+		     "Invalid testing parameters index %u", params_it);
+	current_params_vector = &testing_id_addr_pair_lut[params_it];
+	params_it++;
+}
+
+void test_main(void)
+{
+	reset_global_test_round_params();
+
+	/* This testsuite should setup the conditions for subsequent tests
+	 *   - Checks the initial startup state of the keys pool to be empty
+	 *   - Test bt_keys_get_addr() behaviour when trying to get a key
+	 *     reference for non-existing (ID, Address) pair
+	 */
+	ztest_test_suite(bt_keys_get_addr_get_non_existing_keys,
+			 ztest_unit_test(test_list_is_empty_at_startup),
+			 LISTIFY(CONFIG_BT_MAX_PAIRED, REGISTER_SETUP_TEARDOWN, (,),
```

Comment:
After discussion with @alwa-nordic , we thought it'd be better to explicitly do the iteration over the test vectors in the test cases themselves. It shouldn't be that hard, just a for-loop pretty much.
We don't _need_ one test case result for each vector x testcase. That way the looping is also more explicit for the reader.